### PR TITLE
Docs: Python tutorials doc fixes

### PR DESCRIPTION
**Pull request by TEChopra1000**
## Description ##
* A variety of python tutorial pages has incorrectly formatted hyperlinks, resulting in link-rendering issues in the public docs (e.g. https://mxnet.apache.org/api/python/docs/tutorials/packages/ndarray/sparse/train_gluon.html#Sparse-NDArrays-with-Gluon). I took out the back-slashes surrounding hyperlinked words to fix this issues. 
* Fixed broken links on the Gluon tutorial index page. 
* Small typo fixes

## Testing ##
I tested the doc-build locally and previewed the changes.

### CODE REVIEW

File: docs/python_docs/python/tutorials/extend/custom_layer.md [hunk 1]
```diff
@@ -26,7 +26,6 @@ In this article, I will cover how to create a new layer from scratch, how to use
 
 To create a new layer in Gluon API, one must create a class that inherits from [Block](https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/gluon/block.py#L123) class. This class provides the most basic functionality, and all pre-defined layers inherit from it directly or via other subclasses. Because each layer in Apache MxNet inherits from `Block`, words "layer" and "block" are used interchangeable inside of the Apache MxNet community.
 
-- MXNet [7b24137](https://github.com/apache/incubator-mxnet/commit/7b24137ed45df605defa4ce72ec91554f6e445f0). See Instructions in [Setup and Installation]({{'/get_started'|relative_url}}). 
 The only instance method needed to be implemented is [forward(self, x)](https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/gluon/block.py#L415), which defines what exactly your layer is going to do during forward propagation. Notice, that it doesn't require to provide what the block should do during back propogation. Back propogation pass for blocks is done by Apache MxNet for you.
```
Comment (ehsanmok) [inline docs/python_docs/python/tutorials/extend/custom_layer.md]: Could you point to a specific tag commit when linking to code part for `forward(self, x)`, `hybridize` etc. ?

File: docs/python_docs/python/tutorials/packages/gluon/loss/loss.md [hunk 2]
```diff
@@ -160,7 +160,7 @@ In classification, we often apply the
 softmax operator to the predicted outputs to obtain prediction probabilities,
 and then apply the cross entropy loss against the true labels:
 
-$$ \begin{align}\begin{aligned}p = \softmax({pred})\\L = -\sum_i \sum_j {label}_j \log p_{ij}\end{aligned}\end{align}
+$$ \begin{align}\begin{aligned}p = softmax({pred})\\L = -\sum_i \sum_j {label}_j \log p_{ij}\end{aligned}\end{align}
```
Comment (ehsanmok) [inline docs/python_docs/python/tutorials/packages/gluon/loss/loss.md]: Think latex wise using `\text{softmax}` is better that plain "softmax".

**Comment by vishaalkapoor** [comment]
Modulo Ehsan's comments, approved.

File: docs/python_docs/python/tutorials/packages/gluon/loss/loss.md (hunk 2 above)
Comment (TEChopra1000) [reply docs/python_docs/python/tutorials/packages/gluon/loss/loss.md]: Isn't softmax the name of the function, not a LaTex command? I made this change because a ticket pointed out that there was a LaTex rendering issue here: https://mxnet.apache.org/api/python/docs/tutorials/packages/gluon/loss/loss.html#Cross-Entropy-Loss-with-Softmax

Comment (ehsanmok) [reply docs/python_docs/python/tutorials/packages/gluon/loss/loss.md]: What you did works. Previously it was `\softmax` with the rendering issue, but enclosing it in `\text{softmax}` tells latex to properly handle text written in latex.

**Comment by aaronmarkham** [comment]
@TEChopra1000 please update the title of the PR. These titles end up in the release notes and they should be properly descriptive... like 'documentation update to fix latex rendering" (or similar)

File: docs/python_docs/python/tutorials/extend/custom_layer.md (hunk 1 above)
Comment (TEChopra1000) [inline docs/python_docs/python/tutorials/extend/custom_layer.md]:
```suggestion
The only instance method needed to be implemented is [forward(self, x)](https://github.com/apache/incubator-mxnet/blob/c9818480680f84daa6e281a974ab263691302ba8/python/mxnet/gluon/block.py#L909), which defines what exactly your layer is going to do during forward propagation. Notice, that it doesn't require to provide what the block should do during back propogation. Back propogation pass for blocks is done by Apache MxNet for you. 
```

File: docs/python_docs/python/tutorials/extend/custom_layer.md [hunk 3]
```diff
@@ -26,7 +26,6 @@ In this article, I will cover how to create a new layer from scratch, how to use
 
 To create a new layer in Gluon API, one must create a class that inherits from [Block](https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/gluon/block.py#L123) class. This class provides the most basic functionality, and all pre-defined layers inherit from it directly or via other subclasses. Because each layer in Apache MxNet inherits from `Block`, words "layer" and "block" are used interchangeable inside of the Apache MxNet community.
```
Comment (TEChopra1000) [inline docs/python_docs/python/tutorials/extend/custom_layer.md]:
```suggestion
To create a new layer in Gluon API, one must create a class that inherits from [Block](https://github.com/apache/incubator-mxnet/blob/c9818480680f84daa6e281a974ab263691302ba8/python/mxnet/gluon/block.py#L128) class. This class provides the most basic functionality, and all pre-defined layers inherit from it directly or via other subclasses. Because each layer in Apache MxNet inherits from `Block`, words "layer" and "block" are used interchangeable inside of the Apache MxNet community.
```
